Add service overbooking limits docs

vojtechrojicek opened the pull request:
### Summary

Generate docs for Service overbooking limits.

### Checklist

- [ ] Documentation follows the [Style Guide](https://mews.atlassian.net/wiki/x/KJAoCw)
- [ ] JSON examples updated
- [ ] Properties in JSON examples are in the same order as in property tables
- [x] [Changelog] dated the day when PR merged
- [x] [Changelog] accurately describes all changes
- [x] [Changelog] highlights the affected endpoints or operations
- [x] [Changelog] highlights any deprecations
- [x] All hyperlinks tested
- [ ] [Deprecation Table](https://github.com/MewsSystems/gitbook-connector-api/blob/master/deprecations/README.md) updated if any deprecations
- [x] [SUMMARY.md](https://github.com/MewsSystems/gitbook-connector-api/blob/master/SUMMARY.md) updated if new pages added

[Changelog]: https://github.com/MewsSystems/gitbook-connector-api/blob/master/changelog/README.md

vaclavpacholik commented on operations/serviceoverbookinglimits.md:
```diff
@@ -0,0 +1,208 @@
+
+Care is needed to specify `FirstTimeUnitStartUtc` and `LastTimeUnitStartUtc` in the correct format - see [Datetimes](../guidelines/serialization.md#datetimes).
+
+It's not allowed to update past service overbooking limits outside of `EditableHistoryInterval`.
```
Ok got it, we have it here The maximum size of time interval depends on the service's [time unit](../concepts/time-units.md): 367 days. |

MikeAdamsMews commented on operations/serviceoverbookinglimits.md:
```diff
@@ -0,0 +1,208 @@
+> ### Restricted!
+> This operation is currently in beta-test and as such it is subject to change.
+
+Adds new service overbooking limits with the specified conditions. Note this operation supports [Portfolio Access Tokens](../guidelines/multi-property.md).
```
I would suggest putting these in a single paragraph and a slight change of wording:
```
Adds new service overbooking limits with the specified conditions. Past overbooking limits cannot be updated outside of `EditableHistoryInterval`. Care is needed to specify `FirstTimeUnitStartUtc` and `LastTimeUnitStartUtc` in the correct format - see [Datetimes](../guidelines/serialization.md#datetimes). This operation supports [Portfolio Access Tokens](../guidelines/multi-property.md).
```

MikeAdamsMews commented on operations/serviceoverbookinglimits.md:
```diff
@@ -0,0 +1,208 @@
+
+Care is needed to specify `FirstTimeUnitStartUtc` and `LastTimeUnitStartUtc` in the correct format - see [Datetimes](../guidelines/serialization.md#datetimes).
+
+It's not allowed to update past service overbooking limits outside of `EditableHistoryInterval`.
```
oh, one more thing – I don't think we have a single parameter `EditableHistoryInterval` any more, instead we have the two: 
`AccountingEditableHistoryInterval` and `OperationalEditableHistoryInterval`, not sure which one applies here?

vojtechrojicek replied:
Hello, to be fair, I use `EditableHistoryInterval` because this phrase is also used in `services/updateAvailability` (https://github.com/MewsSystems/gitbook-connector-api/blob/master/operations/services.md#update-service-availability). This way, the terminology is unified when the same backend logic is applied.

So my question is, do I really need to change it?

jnv replied:
So, `EditableHistoryInterval` is available but deprecated, in the code we pick the smaller of `AccountingEditableHistoryInterval` and `OperationalEditableHistoryInterval`.

In this case, I think the service bookings are operational data, so `OperationalEditableHistoryInterval` should be used. The terminology in this case is outdated, we should fix `updateAvailability` as well.

vojtechrojicek replied:
Thanks for clarification, updated.

MikeAdamsMews commented on operations/serviceoverbookinglimits.md:
```diff
@@ -0,0 +1,208 @@
+> ### Restricted!
+> This operation is currently in beta-test and as such it is subject to change.
+
+Deletes service overbooking limits that exactly match the specified conditions. This operation is intended to be used alongside `serviceOverbookingLimits/set`. The specified conditions must be met exactly. The time interval, however, does not need to correspond to an existing service overbooking limit in the system, instead the API uses a splicing algorithm to work out how to divide up any existi ...
```
Same suggestion as above, I prefer a single paragraph for the description. And same comment about `EditableHistoryWindow`. Otherwise looks good.

MikeAdamsMews reviewed (commented):
Just a couple of small suggestions. In addition, can you please update `operations/README.md` which contains a list of all the operations supported. We need to add the new operations with a very brief description. They should also be marked there as **Restricted!**, see other similar examples on the page. Thank you.

vojtechrojicek commented:
@MikeAdamsMews, thank you for all your comments. I have addressed the paragraph changes, added operations to `operations/readme`, and updated the changelog.
I have a question regarding `EditableHistoryInterval`, as this phrase is used in other endpoints with the same logic in place.

Please take a look and share your thoughts or approve if it looks good to you.

vojtechrojicek commented on operations/serviceoverbookinglimits.md:
```diff
@@ -0,0 +1,208 @@
+| `ServiceOverbookingLimits` | array of [ServiceOverbookingLimit](serviceoverbookinglimits.md#serviceoverbookinglimit) | required | Service overbooking limits of the default service. |
+| `Cursor` | string | optional | Unique identifier of the item one newer in time order than the items to be returned. If Cursor is not specified, i.e. null, then the latest or most recent items will be returned. |
+
+#### ServiceOverbookingLimit
```
Added

jnv reviewed (approved):
LGTM, just don't forget to change the changelog date if you're merging it today. ;)
